modules/core/src/lifecycle/prop-types.ts

```diff
@@ -170,6 +177,34 @@ function arrayEqual(array1, array2) {
   return true;
 }
 
+function objectEqual(object1, object2) {
```

Comment:
This method calls `isObject` repeatedly and I suspect has perf issues.

Maybe just use https://github.com/visgl/deck.gl/blob/master/modules/core/src/utils/deep-equal.js ?

Reply:
Switched to using `deepEqual`